File: example/java/config/other/Monster.java
package config.other;

public class Monster {
    private int id;
    private java.util.List<config.Position> posList;
    private int lootId;
    private int lootItemId;
    private config.other.Lootitem RefLoot;
    private config.other.Loot RefAllLoot;

    private Monster() {
    }

    public static Monster _create(configgen.genjava.ConfigInput input) {
        Monster self = new Monster();
        self.id = input.readInt();
        {
            int c = input.readInt();
            if (c == 0) {
                self.posList = java.util.Collections.emptyList();
            } else {
                self.posList = new java.util.ArrayList<>(c);
                for (; c > 0; c--) {
                    self.posList.add(config.Position._create(input));
                }
            }
        }
        self.lootId = input.readInt();
        self.lootItemId = input.readInt();
        return self;
    }

    public int getId() {
        return id;
    }

    public java.util.List<config.Position> getPosList() {
        return posList;
    }

    /**
     * loot
     */
    public int getLootId() {
        return lootId;
    }

    /**
     * item
     */
    public int getLootItemId() {
        return lootItemId;
    }

    public config.other.Lootitem refLoot() {
        return RefLoot;
    }

    public config.other.Loot refAllLoot() {
        return RefAllLoot;
    }

    @Override
    public String toString() {
        return "(" + id + "," + posList + "," + lootId + "," + lootItemId + ")";
    }

    public void _resolveDirect(config.ConfigMgr mgr) {
        RefLoot = mgr.other_lootitem_All.get(new config.other.Lootitem.LootidItemidKey(lootId, lootItemId) );
        java.util.Objects.requireNonNull(RefLoot);
        RefAllLoot = mgr.other_loot_All.get(lootId);
        java.util.Objects.requireNonNull(RefAllLoot);
    }

    public void _resolve(config.ConfigMgr mgr) {
        _resolveDirect(mgr);
    }

    public static Monster get(int id) {
        config.ConfigMgr mgr = config.ConfigMgr.getMgr();
        return mgr.getOtherMonster(id);
    }

    public static java.util.Collection<Monster> all() {
        config.ConfigMgr mgr = config.ConfigMgr.getMgr();
        return mgr.allOtherMonster();
    }

    public static class _ConfigLoader implements config.ConfigLoader {

        @Override
        public void createAll(config.ConfigMgr mgr, configgen.genjava.ConfigInput input) {
            for (int c = input.readInt(); c > 0; c--) {
                Monster self = Monster._create(input);
                mgr.other_monster_All.put(self.id, self);
            }
        }

        @Override
        public void resolveAll(config.ConfigMgr mgr) {
            for (Monster e : mgr.other_monster_All.values()) {
                e._resolve(mgr);
            }
        }

    }

}

File: example/go/config/Other/Monster.go
package other

type Monster struct {
    id int
    posList []config.Position
}

//getters
func (t *Monster) GetId() int {
    return t.id
}

func (t *Monster) GetPosList() []config.Position {
    return t.posList
}

var all []Monster
func GetAll() []Monster {
    return all[:len(all)]:len(all)]
}

var allMap map[int]Monster
func Get(key int) (Monster,bool){
    return allMap[key]
}
        public static Monster Get(int id)
        {
            Monster v;
            return all.TryGetValue(id, out v) ? v : null;
        }

        public static List<Monster> All()
        {
            return all.OrderedValues;
        }

        public static List<Monster> Filter(Predicate<Monster> predicate)
        {
            var r = new List<Monster>();
            foreach (var e in all.OrderedValues)
            {
                if (predicate(e))
                    r.Add(e);
            }
            return r;
        }

        internal static void Initialize(Config.Stream os, Config.LoadErrors errors)
        {
            all = new Config.KeyedList<int, Monster>();
            for (var c = os.ReadInt32(); c > 0; c--) {
                var self = _create(os);
                all.Add(self.Id, self);
            }
        }

